Дипломная работа первая часть

ppetermsc opened the pull request.

ppetermsc committed:
Дипломная работа первая часть

ppetermsc committed:
Обновление структуры: проект перемещен в папку Diplom_1

ppetermsc committed:
Create README.md

okryukova commented on .idea/.gitignore:
⛔️Нужно исправить.  Папку .idea не нужно было загружать в репозиторий. Эта папка должна быть добавлена в .gitignore.

ppetermsc replied:
Исправлено.

okryukova commented on Diplom/Diplom_1/src/test/java/BunTest.java:
```diff
@@ -0,0 +1,42 @@
+
+@RunWith(Parameterized.class)
+public class BunTest {
+
```
⛔️Нужно исправить.  Тестируем только класс Burger

ppetermsc replied:
Исправлено, класс удален.

okryukova commented on Diplom/Diplom_1/src/test/java/BurgerReceiptTest.java:
```diff
@@ -0,0 +1,61 @@
+        String receipt = burger.getReceipt();
+
+        assertNotNull("Чек не должен быть null", receipt);
+        assertTrue("Чек должен содержать название булочки", receipt.contains(bunName));
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

ppetermsc replied:
Исправлено в двух тестовых классах.

okryukova commented on Diplom/Diplom_1/src/test/java/BurgerReceiptTest.java:
```diff
@@ -0,0 +1,61 @@
+
+        assertNotNull("Чек не должен быть null", receipt);
+        assertTrue("Чек должен содержать название булочки", receipt.contains(bunName));
+        assertTrue("Чек должен содержать название ингредиента", receipt.contains(ingredientName.toLowerCase()));
```
⛔️Нужно исправить.  Строку рецепта проверяем целиком, чтобы не пропустить ошибки форматирования

ppetermsc replied:
Исправлено.

okryukova replied:
не исправлено

ppetermsc replied:
Исправлено. Проблема в том, что решая конфликты с Гитом, по ошибке откатил коммиты на один из предыдущих и отправил на проверку старую версию! В новой этот вопрос решен.

okryukova commented on target/classes/praktikum/Bun.class:
⛔️Нужно исправить.  В папке target  оставляем только отчет, остальные файлы нужно удалить из пулл реквеста

ppetermsc replied:
Исправлено.

ppetermsc replied:
Файлы отчета оставлены в папке таргет, остальные удалены.

ppetermsc committed:
temp: save current state

ppetermsc committed:
merge: integrate updated test classes and gitignore

- Use updated BurgerReceiptTest with full receipt validation
- Use updated BurgerTest with single assertion principle
- Remove BunTest as not required by specification
- Add .gitignore for proper file exclusion
- Keep local README.md version

ppetermsc committed:
fix: remove build and IDE files from repository

- Remove target/ folder (build artifacts)
- Remove .idea/ folder (IDE settings)
- Keep only source code and configuration files

okryukova commented on Diplom/Diplom_1/src/test/java/BurgerReceiptTest.java:
```diff
@@ -0,0 +1,61 @@
+
+        String receipt = burger.getReceipt();
+
+        assertNotNull("Чек не должен быть null", receipt);
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

ppetermsc replied:
Исправлено. Отправил старую версию по ошибке.

okryukova commented on Diplom/Diplom_1/src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,86 @@
+        burger.addIngredient(ingredient2);
+        burger.removeIngredient(0);
+
+        assertEquals("Должен остаться один ингредиент после удаления", 1, burger.ingredients.size());
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

ppetermsc replied:
Исправлено в актуальной версии.

okryukova commented on Diplom/Diplom_1/src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,86 @@
+    @Test
+    public void testMoveIngredient() {
+        Burger burger = new Burger();
+        Ingredient ingredient1 = new Ingredient(ingredientType, ingredientName, ingredientPrice);
```
⛔️Нужно исправить. При нейминге не рекомендуется использовать числа (Field2), их еще называют magicNumbers. Очень тяжело поддерживать код с magicNumbers.

okryukova commented on Diplom/Diplom_1/src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,86 @@
+    public void testMoveIngredient() {
+        Burger burger = new Burger();
+        Ingredient ingredient1 = new Ingredient(ingredientType, ingredientName, ingredientPrice);
+        Ingredient ingredient2 = new Ingredient(IngredientType.FILLING, "сосиска", 300f);
```
⛔️Нужно исправить.  Нужно использовать моки

ppetermsc replied:
Сделано.

okryukova commented on src/test/java/BurgerReceiptTest.java:
```diff
@@ -0,0 +1,61 @@
+
+        String receipt = burger.getReceipt();
+
+        assertNotNull("Чек не должен быть null", receipt);
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/BurgerReceiptTest.java:
```diff
@@ -0,0 +1,61 @@
+        assertNotNull("Чек не должен быть null", receipt);
+        assertTrue("Чек должен содержать название булочки", receipt.contains(bunName));
+        assertTrue("Чек должен содержать название ингредиента", receipt.contains(ingredientName.toLowerCase()));
+        assertTrue("Чек должен содержать цену", receipt.contains("Price:"));
```
⛔️Нужно исправить. Тесты дважды в разных пакетах лежат

ppetermsc replied:
Убрал лишние папки из коммита. Мусор оказался там, когда решал конфликты с гитом и откатил не на ту версию.

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,86 @@
+    @Test
+    public void testAddIngredient() {
+        Burger burger = new Burger();
+        Ingredient ingredient = new Ingredient(ingredientType, ingredientName, ingredientPrice);
```
⛔️Нужно исправить.  Нужно моки использовать

ppetermsc replied:
Сделано.

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,86 @@
+    @Test
+    public void testRemoveIngredient() {
+        Burger burger = new Burger();
+        Ingredient ingredient1 = new Ingredient(ingredientType, ingredientName, ingredientPrice);
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

ppetermsc replied:
Сделано.

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,86 @@
+
+    @Test
+    public void testMoveIngredient() {
+        Burger burger = new Burger();
```
⛔️Нужно исправить. Нужна ли в этих тестах параметризация ?

ppetermsc replied:
Оставил только там, где она необходима.

okryukova commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,86 @@
+        assertNotNull("Булочка должна быть выбрана", burger.bun);
+        assertEquals("Булочка должна совпадать", bun, burger.bun);
+    }
+}
```
⛔️Нужно исправить. Нет отчёта о тестировании

ppetermsc replied:
Отчет в папке таргет, остальные файлы из коммита убрал.

ppetermsc committed:
Почистил репозиторий, сделал необходимые изменения в проекте, добавил файлы отчета JaCoCo

ppetermsc committed:
fix: optimize test classes

ppetermsc committed:
chore: remove .DS_Store system files

okryukova commented on target/site/jacoco/index.html:
⛔️Нужно исправить. Нужно приложить все файлы из папки jacoco

ppetermsc replied:
ОК. Сделано.

ppetermsc committed:
docs: add complete JaCoCo report resources
